add 'runWithClient' example to http-integration.mdx (#10979)
getsentry/sentry-docs: docs/platforms/dart/integrations/http-integration.mdx

sidebar_order: 2
---

## Using the `runWithClient` function

> With [`runWithClient`](https://pub.dev/documentation/http/latest/http/runWithClient.html) you can define a global `SentryHttpClient` and run it in its own [`Zone`](https://api.dart.dev/stable/3.4.4/dart-async/Zone-class.html).

Reviewer: ```suggestion
With [`runWithClient`](https://pub.dev/documentation/http/latest/http/runWithClient.html), you can define a global `SentryHttpClient` and run it in its own [`Zone`](https://api.dart.dev/stable/3.4.4/dart-async/Zone-class.html).
```

> With [`runWithClient`](https://pub.dev/documentation/http/latest/http/runWithClient.html) you can define a global `SentryHttpClient` and run it in its own [`Zone`](https://api.dart.dev/stable/3.4.4/dart-async/Zone-class.html).

Reviewer 2: Since you added the code snippet as a tab, now I don't think this runWithClient section is necessary anymore. Maybe you could add a short introduction/summary of the two alternatives the user could do to use the http integration. One is using the SentryHttpClient directly and the other is using the runWithClient function. Or something in that direction

Author: I rephrased this section. Please have a look at my latest commit.

> Depending on your use case, you can either use the `SentryHttpClient` directly and call its methods (see the `default` tab), or you can use the [`runWithClient`](https://pub.dev/documentation/http/latest/http/runWithClient.html) function (see the `runWithClient` tab). With [`runWithClient`](https://pub.dev/documentation/http/latest/http/runWithClient.html), a fresh instance of the `client` is used and the other instances are not affected. It also runs in a separate [`Zone`](https://api.dart.dev/stable/3.5.0/dart-async/Zone-class.html), allowing you to override the functionality of the existing [`Zone`](https://api.dart.dev/stable/3.5.0/dart-async/Zone-class.html).

Reviewer: ```suggestion
You can use the `SentryHttpClient` and call its methods directly, or you can use it with the [`runWithClient`](https://pub.dev/documentation/http/latest/http/runWithClient.html) function. If you need to use a fresh instance of the `client` (so that other instances are not affected) or to run in a separate [`Zone`](https://api.dart.dev/stable/3.5.0/dart-async/Zone-class.html), which allows you to override the functionality of the existing [`Zone`](https://api.dart.dev/stable/3.5.0/dart-async/Zone-class.html), then use the `runWithClient` function (see the `runWithClient` tab). Otherwise, just use `SentryHttpClient` directly and call its methods (see the `default` tab).
```

await transaction.finish(status: SpanStatus.ok());
```

> ## Using the `runWithClient` function

Reviewer 2: Since this is a way to 'initialize' the integration, we can maybe push this up to the very top.

Maybe we can just add the code snippet as a tab like this: https://docs.sentry.io/platforms/kotlin-multiplatform/initialization-strategies/#usage

Author: moved it now to the top.
